Reload rsyslog after installing node-agent

TimothyAsirJeyasing opened the pull request:
The patch does the following things after installing node agent.
 * Start rsyslogd service if its not started
 * Reload rsyslogd after tendrl-node-agent service restart

tendrl-bug-id: Tendrl/tendrl-ansible#683
Signed-off-by: Timothy Asir J <tjeyasin@redhat.com>

TimothyAsirJeyasing committed:
Reload rsyslog after installing node-agent

The patch does the following things after installing node agent.
 * Start rsyslogd service if its not started
 * Reload rsyslogd after tendrl-node-agent service restart

tendrl-bug-id: Tendrl/tendrl-ansible#683
Signed-off-by: Timothy Asir J <tjeyasin@redhat.com>

TimothyAsirJeyasing commented:
@mbukatov , @r0h4n , @nthomas-redhat , @shtripat Please review

shtripat commented:
@TimothyAsir kindly rebase this PR

cybernth commented on roles/tendrl-storage-node/handlers/main.yml:
```diff
@@ -10,3 +10,8 @@
   service:
     name=tendrl-node-monitoring
```
Tendrl-node-monitoring is nor a tendrl service anymore. This needs to be cleaned up

TimothyAsirJeyasing replied:
I am not doing anything with tendrl-node-monitoring. This will go away after i resolve merge conflict. I will rebase and update the PR so that this will be corrected.

cybernth commented on roles/tendrl-storage-node/tasks/tendrl-node-agent.yml:
```diff
@@ -22,6 +22,8 @@
     - restart tendrl-node-agent
   when: tendrl_gluster_provisioning_support == True
   run_once: true
+  notify:
```
Have you tested this?
By looking at the code, this block gets executed only if the tendrl_gluster_provisioning_support == True; which is not supported now

TimothyAsirJeyasing replied:
Yes, i agree, this will get corrected automatically when i resolve the merge conflict.

TimothyAsirJeyasing committed:
Reload rsyslog after installing node-agent

The patch does the following things after installing node agent.
* Start rsyslogd service if its not started
* Reload rsyslogd after tendrl-node-agent service restart

tendrl-bug-id: Tendrl/tendrl-ansible#683
Signed-off-by: Timothy Asir J <tjeyasin@redhat.com>

TimothyAsirJeyasing commented:
I have rebased and updated the pr. Kindly review

cybernth commented on roles/tendrl-storage-node/tasks/tendrl-node-agent.yml:
```diff
@@ -33,6 +35,8 @@
     - restart tendrl-node-agent
   when: etcd_tls_client_auth|bool == True
+  notify:
+    - reload rsyslog
```
I don't think this is required. The code at line no 20-21 does the job for you?

cybernth commented:
@TimothyAsirJeyasing , Looks like reload doesnot work some distributions, so you have to restart and don't use reload. FYI https://github.com/Tendrl/documentation/issues/97

mbukatov commented on roles/tendrl-storage-node/tasks/tendrl-node-agent.yml:
```diff
@@ -17,6 +17,7 @@
       line: "graphite_host: {{ graphite_fqdn }}"
   notify:
     - restart tendrl-node-agent
+    - reload rsyslog
```
I would rather add this `reload rsyslog` notification to `Install tendrl-node-agent` task. Reloading rsyslog config is needed because of default config file shipped in the package, and has nothing to do with it's configuration.

TimothyAsirJeyasing replied:
No state change observed in Install tendrl-node-agent which does not initiate the notification because the package gets installed by "Update all installed packages" itself. I will find the other suitable place and move the notification

mbukatov replied:
Good point.

mbukatov commented on roles/tendrl-storage-node/tasks/main.yml:
```diff
@@ -22,3 +22,4 @@
     state=latest
 
 - include: tendrl-node-agent.yml
+- include: rsyslog.yml
```
It may be better to include `rsyslog.yml` before including `tendrl-node-agent.yml`. That said, this have no impact on triggering rsyslogd restart via notification.

TimothyAsirJeyasing replied:
+1

mbukatov commented on roles/tendrl-storage-node/tasks/rsyslog.yml:
```diff
@@ -0,0 +1,6 @@
+- name: ensure rsyslog is running
+  service:
+    name=rsyslogd
+    state=started
```
Here we are going a bit deeper than originally intended. While this makes sense, shouldn't we make sure that the service is enabled as well? It's on the same level as making sure that the rsyslog is running.

TimothyAsirJeyasing replied:
i will add enabled=yes ?

mbukatov reviewed (changes requested):
Besides comments below, I think that we need to add the same for tendrl-server role, because node-agent is installed there as well.

That said, after finishing this review, I'm more inclined to do this via rpm postscript, as:

* we need to make sure that rsyslog is restarted in both roles, as node-agent is installed there (in ansible, this duplication would be best addressed via rsyslog role)
* when we update rsyslog config or fix a bug there, nobody will restart rsyslog for the change to take effect

That said, let's keep moving there, unless we have additional good reason to switch to rpm scriptlet solution.

TimothyAsirJeyasing committed:
Restart rsyslog after installing node-agent

The patch does the following things after installing node agent.
* Start rsyslogd service if its not started
* Restart rsyslogd after installing tendrl-node-agent

Signed-off-by: Timothy Asir J <tjeyasin@redhat.com>

TimothyAsirJeyasing commented:
@r0h4n , @nthomas-redhat , @shtripat , @mbukatov , Please review

cybernth commented on roles/tendrl-server/tasks/tendrl-node-agent.yml:
```diff
@@ -52,3 +54,5 @@
   service:
     name=tendrl-node-agent
     state=started
+  notify:
```
Do we need this?

TimothyAsirJeyasing replied:
yes we need this

mbukatov replied:
Why is that?

mbukatov replied:
Is this a workaround for: https://github.com/Tendrl/tendrl-ansible/pull/69#discussion_r168075917 ?

mbukatov replied:
If this is a workaround for: #69 (comment), I propose to add a note about this into a comment there and merge it as it is now.

cybernth commented on roles/tendrl-storage-node/tasks/tendrl-node-agent.yml:
```diff
@@ -44,3 +45,5 @@
   service:
     name=tendrl-node-agent
     state=started
+  notify:
```
Same as earlier comment

TimothyAsirJeyasing replied:
Yes its required. We can have any number of notify but only one will be executed at the end.
The install node agent return no state change most of the time. If package not found or package version is old then also it does not provide a state change because the update package happens at top level for all the packages. So we need this here!. However in some cased we need for install node agent also.

cybernth commented on roles/tendrl-storage-node/tasks/rsyslog.yml:
```diff
@@ -0,0 +1,7 @@
+---
```
This is defined in tedrl-storage-node and used in tendrl-server?
You need to have this at both places right?

TimothyAsirJeyasing replied:
Yes, During the test i used. i forgot to git add.  I will add the file

TimothyAsirJeyasing replied:
Done

TimothyAsirJeyasing committed:
Restart rsyslog after installing node-agent

The patch does the following things after installing node agent.
* Start rsyslogd service if its not started
* Restart rsyslogd after installing tendrl-node-agent

Signed-off-by: Timothy Asir J <tjeyasin@redhat.com>

cybernth commented on roles/tendrl-server/tasks/rsyslog.yml:
```diff
@@ -0,0 +1,7 @@
+---
+
+- name: ensure rsyslog is enabled and running SERVER-NGCHECKINGCHECKINGCHECKINGCHECKINGCHECKING
```
What is this long string?

TimothyAsirJeyasing replied:
Typo :) I add this line to quickly check the debug message. I will remove this string.

cybernth reviewed (changes requested):
Is this tested?

TimothyAsirJeyasing committed:
Restart rsyslog after installing node-agent

The patch does the following things after installing node agent.
* Start rsyslogd service if its not started
* Restart rsyslogd after installing tendrl-node-agent

Signed-off-by: Timothy Asir J <tjeyasin@redhat.com>

cybernth reviewed (approved):
LGTM

mbukatov committed:
add a note about workaround
